File: src/test/java/kim/present/solve/programmers/Solution250136Test.java
package kim.present.solve.programmers;

import org.junit.jupiter.api.Test;

import static kim.present.solve.programmers.ProgrammersTester.copy;
import static org.junit.jupiter.api.Assertions.assertEquals;

class Solution250136Test {

    static Solution250136 javaSolution = new Solution250136();
    static Solution250136kt kotlinSolution = new Solution250136kt();

    @Test
    void testcase_1() {
        int[][] land = {
                {0, 0, 0, 1, 1, 1, 0, 0},
                {0, 0, 0, 0, 1, 1, 0, 0},
                {1, 1, 0, 0, 0, 1, 1, 0},
                {1, 1, 1, 0, 0, 0, 0, 0},
                {1, 1, 1, 0, 0, 0, 1, 1}
        };
        assertEquals(9, javaSolution.solution(ProgrammersTester.copy(land)));
        assertEquals(9, kotlinSolution.solution(ProgrammersTester.copy(land)));
    }

    @Test
    void testcase_2() {
        int[][] land = {
                {1, 0, 1, 0, 1, 1},
                {1, 0, 1, 0, 0, 0},
                {1, 0, 1, 0, 0, 1},
                {1, 0, 0, 1, 0, 0},
                {1, 0, 0, 1, 0, 1},
                {1, 0, 0, 0, 0, 0},
                {1, 1, 1, 1, 1, 1}
        };
        assertEquals(16, javaSolution.solution(ProgrammersTester.copy(land)));
        assertEquals(16, kotlinSolution.solution(ProgrammersTester.copy(land)));
    }

    @Test
    void testcase_3() {
        int[][] land = {
                {0, 0, 0, 1, 1, 1, 1, 1},
                {0, 0, 0, 0, 0, 0, 0, 1},
                {1, 1, 1, 1, 1, 1, 1, 1},
                {1, 1, 1, 0, 0, 0, 0, 0},
                {1, 1, 1, 0, 0, 0, 1, 1}
        };
        assertEquals(22, javaSolution.solution(ProgrammersTester.copy(land)));
        assertEquals(22, kotlinSolution.solution(ProgrammersTester.copy(land)));
    }

    @Test
    void testcase_4() {
        int[][] land = new int[500][500];
        for (int y = 0; y < 500; y++) {
            for (int x = 0; x < 500; x++) {
                land[y][x] = 1;
            }
        }
        assertEquals(500 * 500, javaSolution.solution(ProgrammersTester.copy(land)));
        assertEquals(500 * 500, kotlinSolution.solution(ProgrammersTester.copy(land)));
    }
}

File: src/test/java/kim/present/solve/programmers/Solution250136Test.kt
package kim.present.solve.programmers

import org.junit.jupiter.api.TestFactory

internal class Solution250136Test : TestBase1<Array<IntArray>, Int>() {
    init {
        +Solution250136()::solution
        +Solution250136kt()::solution
    }

    @TestFactory
    fun run() = test {
        {
            of(
                of(0, 0, 0, 1, 1, 1, 0, 0),
                of(0, 0, 0, 0, 1, 1, 0, 0),
                of(1, 1, 0, 0, 0, 1, 1, 0),
                of(1, 1, 1, 0, 0, 0, 0, 0),
                of(1, 1, 1, 0, 0, 0, 1, 1)
            )
        } expected 9

        {
            of(
                of(1, 0, 1, 0, 1, 1),
                of(1, 0, 1, 0, 0, 0),
                of(1, 0, 1, 0, 0, 1),
                of(1, 0, 0, 1, 0, 0),
                of(1, 0, 0, 1, 0, 1),
                of(1, 0, 0, 0, 0, 0),
                of(1, 1, 1, 1, 1, 1)
            )
        } expected 16

        {
            of(
                of(0, 0, 0, 1, 1, 1, 1, 1),
                of(0, 0, 0, 0, 0, 0, 0, 1),
                of(1, 1, 1, 1, 1, 1, 1, 1),
                of(1, 1, 1, 0, 0, 0, 0, 0),
                of(1, 1, 1, 0, 0, 0, 1, 1)
            )
        } expected 22

        { Array(500) { IntArray(500) { 1 } } } expected 500 * 500
    }
}
